File: golang/1.array/find_min_wating_time.go
package main

import "sort"

func FindMinimumWatingTime(q []int) int {
	sort.Ints(q)
	wt := 0

	for I, d := range q {
		ql := len(q) - I - 1
		wt += ql * d
	}

	return wt
}

File: ts/1.array/find-min-wating-time.ts
// o(nlogn)
export function findMinimumWatingTime(q : number[]) :number { 
    q.sort((a:number , b:number) => a > b ? 1 : -1 );
    let wt = 0 ; 

    for (let i=0 ; i<q.length ; i++) { 
        let ql = q.length - i - 1 ; 
        wt += q[i] * ql ;
    }

    return wt;
}
